edge/pkg/edged/edged.go

```diff
@@ -1378,14 +1378,27 @@ func (e *edged) deletePod(obj interface{}) {
 
 func (e *edged) getSecretsFromMetaManager(pod *v1.Pod) ([]v1.Secret, error) {
 	var secrets []v1.Secret
-	for _, imagePullSecret := range pod.Spec.ImagePullSecrets {
-		secret, err := e.metaClient.Secrets(pod.Namespace).Get(imagePullSecret.Name)
-		if err != nil {
-			return nil, err
+	if pod.Spec.ImagePullSecrets == nil{
+		for _, containers := range pod.Spec.Containers{
+			for _, env := range containers.Env{
+				secret, err := e.metaClient.Secrets(pod.Namespace).Get(env.ValueFrom.SecretKeyRef.Name)
+				if err != nil {
+					return nil, err
+				}
+				secrets = append(secrets, *secret)
+			}
+		}
+	}else{
+		for _, imagePullSecret := range pod.Spec.ImagePullSecrets {
+			secret, err := e.metaClient.Secrets(pod.Namespace).Get(imagePullSecret.Name)
+			if err != nil {
+				return nil, err
+			}
+			secrets = append(secrets, *secret)
 		}
-		secrets = append(secrets, *secret)
 	}
 
+
 	return secrets, nil
```

Comment:
This func is only used to get the pull secret, we can renamed it to `getPullSecretsFromMetaManager`.

Reply:
Thank you for your help. I will renamed it.